src/core/decode_stream.js

```diff
@@ -22,6 +22,28 @@ import { Stream } from "./stream.js";
 // buffer.
 const emptyBuffer = new Uint8Array(0);
 
+function makeLengthBigEnough(startLength, minLength) {
+  // Compute the first power of two that is as big as y.
+
+  // This function is a bit tricky but always faster than:
+  // while (startLength < minLength) startLength *= 2;
+  // The bigger is the ratio, the faster is the tricky version compared the
+  // while loop.
```

Comment:
The entire comment is most helpful to explain this function, thanks for that!

---

Some grammar nits on these specific lines:
```suggestion
  // The bigger the ratio is, the faster the tricky version is compared to
  // the while loop.
```